addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -526,9 +535,18 @@ private List<Transfer> generateNPMContentsFromTransfer( final Transfer transfer,
             tarballTarget = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), tarballPath ) );
 
         }
+        catch ( final StreamConstraintsException e )
+        {
+            // Refer to https://www.javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.15.2/com/fasterxml/jackson/core/exc/StreamConstraintsException.html
+            logger.error( String.format(
+                    "[NPM] Json node parse failed due to violation of Jackson stream constraint: %s. Reason: %s",
+                    resource, e.getMessage() ), e );
+            throw new StreamConstraintsException( "Violation of Jackson stream constraint." );
```

Comment:
I think you can directly throw the exception instead of wrap it again here.

Reply:
@ligangty Yeah, done.

Reply:
What I mean is no need to do this "new StreamConstraintsException("xxxx")" but just "throw e". This Exception should still be caught before the next "catch(IOException)" because it is also an IOException. If you don't catch it separately, it will also be caught by next catch and do a single logging but no throw there.

Reply:
@ligangty done, thanks.